### DISCUSSION TITLE
How can I to install torch with a particular CUDA

### DISCUSSION THREAD

**Discussion by bzorzet98** [python-poetry/poetry#8731] [Q&A]
I am new in Poetry, and I want to use for my PhD project. 

So, I try to install Pytorch 2.1.1 for Linux and CUDA 11.8. I runned this line in terminal : poetry add torch torchvision torchaudio --url https://download.pytorch.org/whl/cu118

And obtain this issues: 
The Poetry configuration is invalid:
  - [dependencies.torch] {'version': '*', 'index': 'https://download.pytorch.org/whl/cu118'} is not valid under any of the given schemas
  - [dependencies.torchaudio] {'version': '*', 'index': 'https://download.pytorch.org/whl/cu118'} is not valid under any of the given schemas
  - [dependencies.torchvision] {'version': '*', 'index': 'https://download.pytorch.org/whl/cu118'} is not valid under any of the given schemas

**Comment by dimbleby** [comment]
Please link to the section of the documentation that made you think this command would work

**Comment by abn** [comment]
```sh
poetry source add --priority=explicit pytorch-gpu https://download.pytorch.org/whl/cu118

poetry add --source pytorch-gpu torch torchvision torchaudio
```

Should work.

**Reply by parthmahajan008** [reply]
Thanks man, was looking for this for so long!!
